hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/SqlClientPool.java

```diff
@@ -266,81 +266,91 @@ public static URI parse(String url) {
 
 	private class ProxyConnection implements ReactiveConnection {
 		private ReactiveConnection connection;
-
-		CompletionStage<ReactiveConnection> connection() {
-			return connection == null ?
-					getConnection().thenApply(conn -> connection = conn) :
-					CompletionStages.completedFuture(connection);
+		private boolean connected;
+
+		private <T> CompletionStage<T> withConnection(Function<ReactiveConnection,CompletionStage<T>> operation) {
+			if (!connected) {
+				connected = true;
+				return getConnection()
```

Comment:
Shouldn't you set `connected` to true only after you set connection (o in the same CompletionStage)? Could it cause some issues if `getConnection()` takes a long time before assigning the `connection`?

Reply:
The whole point of having a separate flag there is so that I can detect an illegal "concurrent" request for a connection, and throw an exception instead of obtaining a second connection.

Reply:
ok